### PR TITLE
Handle Node Removal from Discovery correctly

### LINKED ISSUE #197: Unable to remove a node from the cluster
The ```Watch``` method of the discovery backend calls ```cluster.UpdateNodes``` method to update the  node list according the return value of the ```Fetch``` method of the discovery service.
But it only inserts the nodes that aren't exist in the node list and doesn't touch the nodes that are "removed" from the discovery service.

If a node is removed from the discovery service, the ```swarm list``` command returns the correct result because it calls the ```Fetch``` method of the discovery service directly. But in fact the removed node is still in the node list of the cluster.

### PR DESCRIPTION
**Pull request by aluzzardi**
Fixes #197

### CODE REVIEW

**Comment by aluzzardi** [comment]
ping @vieux @abronan

File: api/router.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ func createRouter(c *context, enableCors bool) *mux.Router {
 			localRoute := route
 			localFct := fct
 			wrap := func(w http.ResponseWriter, r *http.Request) {
-				log.WithFields(log.Fields{"method": r.Method, "uri": r.RequestURI}).Info("HTTP request received")
```
Comment (vieux) [inline api/router.go]: ok

**Comment by vieux** [comment]
LGTM

**Comment by vieux** [comment]
ping @abronan

**Comment by abronan** [comment]
LGTM

**Pull request merged**